R/bootstrap.R

```diff
@@ -827,3 +887,14 @@ renv_bootstrap_user_dir_impl <- function() {
 
 }
 
+renv_bootstrap_version_is_dev <- function(version) {
+  # if the renv version number is a sha, or has 4 components, it must
+  # be retrieved via github
+  if (!grepl("[.-]", version)) {
+    # not . or -, so must be a sha
+    TRUE
+  } else {
+    components <- strsplit(version, "[.-]")[[1]]
```

Comment:
Or could check `length(components) != 3`? Since we'll always use 3 component version strings for CRAN releases.

Reply:
Sure; we should probably check one way everywhere 😄